File: backend/src/main/java/com/photo/entity/Album.java
package com.photo.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.io.Serializable;
import java.time.LocalDateTime;

@Data
@TableName("t_album")
public class Album implements Serializable {
    @TableId(type = IdType.AUTO)
    private Long id;

    private Long userId;

    private String name;

    private String description;

    private Long coverPhotoId;

    private Integer photoCount;

    private Integer isPublic;

    private Integer sortOrder;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
}

File: frontend/src/types/album.ts
export interface Album {
  id: number
  userId: number
  name: string
  description: string
  coverPhotoId: number
  coverPhotoUrl: string
  photoCount: number
  isPublic: number
  sortOrder: number
  createdAt: string
  updatedAt: string
}

export interface AlbumDTO {
  id?: number
  name: string
  description?: string
  coverPhotoId?: number
  isPublic?: number
}
